integration-test/common/conftest.py

```diff
@@ -1,8 +1,41 @@
+""" This contains pytest hook implementations.
+
+pytest will look for a conftest.py for directory-specific pytest hook
+implementations (tests in the same directory will share the hooks). All the hook
+functions have a pytest_ prefix, tests will invoke the hooks defined in
+conftest.py. Refer to https://docs.pytest.org/en/stable/reference.html for more
+details about the pytest APIs.
+
+Also a bit introduction on pytest-xdist. It's a plugin to pytest that extends
+pytest mainly for parallel test run. Simply use the "-n" command line option to
+specify the number of processes to run the tests. xdist uses a master node for
+control process and worker nodes for running actual tests. Each node will start
+a pytest session to collect the tests (they must see the tests consistent
+across all nodes), and then the tests will be distributed to all the workers.
+
+Here we implement these pytest hooks:
```

Comment:
These comments are fantastic, but I think they would belong better as documentation under the actual functions themselves, rather than up here.

Reply:
Right, moved these to the actual functions themselves.